controllers/topology/reconcile_state.go

```diff
@@ -136,6 +153,7 @@ func (r *ClusterReconciler) reconcileCluster(ctx context.Context, s *scope.Scope
 	if err := patchHelper.Patch(ctx); err != nil {
 		return errors.Wrapf(err, "failed to patch %s", tlog.KObj{Obj: s.Current.Cluster})
 	}
+	r.recorder.Eventf(s.Current.Cluster, corev1.EventTypeNormal, "TopologyUpdate", "Updated %q", tlog.KObj{Obj: s.Current.Cluster})
```

Comment:
I'm not sure about the conventions around event reasons, but it looks like in other cases we often have `Failed<>` and a `Successful<>` event.

This could mean:
* let's prefix the reason with Successful or Failed
* let's also publish an event in the error case

Reply:
Also not sure if it's better to use something like `SuccessfulTopologyUpdate` or `SuccessfulClusterUpdate`
(I'm fine with both, would be interesting if there are some Kubernetes upstream conventions)

Reply:
> Also not sure if it's better to use something like SuccessfulTopologyUpdate or SuccessfulClusterUpdate
(I'm fine with both, would be interesting if there are some Kubernetes upstream conventions)

I'm personally -1 with this one, given that there is already event type normal/warning
Also if I look at Kubernetes events after creating a cluster, Successful/Failed prefix is not usual and the only place where this is usual are MachineDeployment/MachineSet :-(

```
6m3s        Normal   Starting                  node/capi-test-control-plane                  Starting kubelet.
6m3s        Normal   NodeHasSufficientMemory   node/capi-test-control-plane                  Node capi-test-control-plane status is now: NodeHasSufficientMemory
6m3s        Normal   NodeHasNoDiskPressure     node/capi-test-control-plane                  Node capi-test-control-plane status is now: NodeHasNoDiskPressure
6m3s        Normal   NodeHasSufficientPID      node/capi-test-control-plane                  Node capi-test-control-plane status is now: NodeHasSufficientPID
6m3s        Normal   NodeAllocatableEnforced   node/capi-test-control-plane                  Updated Node Allocatable limit across pods
5m51s       Normal   Starting                  node/capi-test-control-plane                  Starting kubelet.
5m51s       Normal   NodeHasSufficientMemory   node/capi-test-control-plane                  Node capi-test-control-plane status is now: NodeHasSufficientMemory
5m51s       Normal   NodeHasNoDiskPressure     node/capi-test-control-plane                  Node capi-test-control-plane status is now: NodeHasNoDiskPressure
5m51s       Normal   NodeHasSufficientPID      node/capi-test-control-plane                  Node capi-test-control-plane status is now: NodeHasSufficientPID
5m51s       Normal   NodeAllocatableEnforced   node/capi-test-control-plane                  Updated Node Allocatable limit across pods
5m39s       Normal   RegisteredNode            node/capi-test-control-plane                  Node capi-test-control-plane event: Registered Node capi-test-control-plane in Controller
5m34s       Normal   Starting                  node/capi-test-control-plane                  Starting kube-proxy.
5m21s       Normal   NodeReady                 node/capi-test-control-plane                  Node capi-test-control-plane status is now: NodeReady
5s          Normal   SuccessfulCreate          machineset/my-cluster1-md1-rdtlv-864ffd466c   Created machine "my-cluster1-md1-rdtlv-864ffd466c-hb74t"
5s          Normal   SuccessfulCreate          machinedeployment/my-cluster1-md1-rdtlv       Created MachineSet "my-cluster1-md1-rdtlv-864ffd466c"
5s          Normal   TopologyCreate            cluster/my-cluster1                           Created "DockerCluster/my-cluster1-fmjqk"
5s          Normal   TopologyCreate            cluster/my-cluster1                           Created "DockerMachineTemplate/my-cluster1-control-plane-shgx4"
5s          Normal   TopologyCreate            cluster/my-cluster1                           Created "KubeadmControlPlane/my-cluster1-nzmzw"
5s          Normal   TopologyUpdate            cluster/my-cluster1                           Updated "Cluster/my-cluster1"
5s          Normal   TopologyCreate            cluster/my-cluster1                           Created "DockerMachineTemplate/my-cluster1-md1-infra-x5dgv"
5s          Normal   TopologyCreate            cluster/my-cluster1                           Created "KubeadmConfigTemplate/my-cluster1-md1-bootstrap-x29kb"
5s          Normal   TopologyCreate            cluster/my-cluster1                           Created "MachineDeployment/my-cluster1-md1-rdtlv"
```

> let's also publish an event in the error case

I'm +1 but I would defer to a follow up PR given that I would like to have consistency with conditions